scripts/models/TownHall.js

```diff
@@ -132,29 +119,61 @@
   };
 
   TownHall.lookupReps = function (key, value) {
-    if (key === 'zip') {
-      var representativePromise = $.ajax({
-        url: 'https://congress.api.sunlightfoundation.com/legislators/locate?' + key + '=' + value,
-        dataType: 'jsonp'
-      });
-    } else if (key === 'state') {
-      var representativePromise = $.ajax({
-        url: 'https://congress.api.sunlightfoundation.com/legislators?state=' + value + '&chamber=senate',
-        dataType: 'jsonp'
+    return _lookupRepIds(key, value).then(function(keys) {
+      return _lookupRepObjs(keys);
+    });
+  };
+
+  function _lookupRepObjs(keys) {
+    return MoC.all.then(function(MoCs) {
+      return keys.map(function(key) {
+        return MoCs.find(function(member) {
+          return member.govtrack_id === key;
```

Comment:
It seems since we have the endpoint, it makes sense to do another firebase lookup here using the govtrack id. Even though we have all the mocs loaded on the front end, it avoids iterating over an pretty big array, and the single mapping lookup is relatively fast.